compiler/src/dmd/dsymbolsem.d

```diff
@@ -1169,10 +1169,12 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
                     // https://issues.dlang.org/show_bug.cgi?id=14166
                     // Don't run CTFE for the temporary variables inside typeof
                     dsym._init = dsym._init.initializerSemantic(sc, dsym.type, sc.intypeof == 1 ? INITnointerpret : INITinterpret);
+                    import dmd.semantic2 : lowerStaticAAs;
+                    lowerStaticAAs(dsym, sc);
```

Comment:
This is becoming a worrying pattern, particularly as failure to call `lowerStaticAAs` after initializer semantic results in an ICE.

#15683

Reply:
The problem is that it should only happen to variables in the data segment, and initializerSemantic doesn't have that information